Library/Keychain.swift

```diff
@@ -0,0 +1,115 @@
+import Foundation
+import Security
+
+public enum KeychainError: Error {
+  case unexpectedStatus(OSStatus)
+  case unableToDecodePasswordData
+
+  public var errorMessage: String {
+    switch self {
+    case let .unexpectedStatus(status):
+      let errString = SecCopyErrorMessageString(status, nil) as String?
+      return errString ?? "Unknown keychain error"
+    case .unableToDecodePasswordData:
+      return "Unable to decode password data from keychain."
+    }
+  }
+}
+
+public struct Keychain {
+  private static var serviceName: String {
+    return Bundle.main.bundleIdentifier ?? "com.kickstarter.kickstarter"
+  }
+
+  public static func deleteAllPasswords() throws {
+    let query: [String: AnyObject] = [
+      kSecAttrService as String: self.serviceName as AnyObject,
+      kSecClass as String: kSecClassGenericPassword as AnyObject
+    ]
+
+    let status = SecItemDelete(query as CFDictionary)
+
+    if status == errSecItemNotFound {
+      return
+    } else if status != errSecSuccess {
+      throw KeychainError.unexpectedStatus(status)
+    }
+  }
+
+  public static func deletePassword(forAccount account: String) throws {
+    let query: [String: AnyObject] = [
+      kSecAttrService as String: self.serviceName as AnyObject,
+      kSecClass as String: kSecClassGenericPassword as AnyObject,
+      kSecAttrAccount as String: account as AnyObject
+    ]
+
+    let status = SecItemDelete(query as CFDictionary)
+
+    if status == errSecItemNotFound {
+      return
+    } else if status != errSecSuccess {
+      throw KeychainError.unexpectedStatus(status)
+    }
+  }
+
+  public static func storePassword(_ password: String, forAccount account: String) throws {
```

Comment:
Probably worth noting - I decided to keep the nomenclature of this code fairly close to the way keychain actually works, which uses "accounts" and "passwords". In our case, we're going to be storing an oAuth token against some bit of user data (probably a user e-mail or a user ID). I plan on writing a quick little extension when we integrate this code with the rest of the app to make that clearer:

```
extension Keychain {
  public static func storeOAuthToken(_ token: String, forUserID id: Int) { 
     self.storePassword(token, forAccount: "\(id)")
  }
}
```